api/config/environment.go

```diff
@@ -89,16 +89,20 @@ type ResourceRequestConfig struct {
 	MemoryRequest string `yaml:"memory_request"`
 }
 
-func initEnvironmentConfigs(path string) []EnvironmentConfig {
+func InitEnvironmentConfigs(path string) []EnvironmentConfig {
 	cfgFile, err := os.ReadFile(path)
 	if err != nil {
 		log.Panicf("unable to read deployment config file: %s", path)
 	}
 	var configs []EnvironmentConfig
-	err = yaml.Unmarshal(cfgFile, &configs)
-	if err != nil {
+	if err = yaml.Unmarshal(cfgFile, &configs); err != nil {
 		log.Panicf("unable to unmarshall deployment config file:\n %s,\nDue to: %v", cfgFile, err)
 	}
+	for _, env := range configs {
+		if env.K8sConfig == nil {
+			log.Panicf("Error, k8sConfig for %s is nil", env.Name)
+		}
+	}
```

Comment:
This new check ensures that all the k8s credentials configs are not nil (a similar check exists in Turing).